oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/MappedJwtClaimSetConverter.java

```diff
@@ -0,0 +1,232 @@
+	private static final Converter<Object, Instant> TEMPORAL_CONVERTER = new TemporalConverter();
+
+	private final Map<String, Converter<Object, ?>> attributeConverters = new HashMap<>();
+
+	/**
+	 * Constructs a {@link MappedJwtClaimSetConverter} with the provided arguments
+	 *
+	 * This will completely replace any set of default converters.
+	 *
+	 * @param attributeConverters The {@link Map} of converters to use
+	 */
+	public MappedJwtClaimSetConverter(Map<String, Converter<Object, ?>> attributeConverters) {
+		Assert.notNull(attributeConverters, "attributeConverters cannot be null");
+		this.attributeConverters.putAll(attributeConverters);
+	}
+
+	/**
+	 * Construct a {@link MappedJwtClaimSetConverter}, overriding individual attribute
+	 * converters with the provided {@link Map} of {@link Converter}s.
+	 *
+	 * For example, the following would give an instance that is configured with only the default
+	 * attribute converters:
+	 *
+	 * <pre>
+	 * 	MappedJwtClaimSetConverter.withDefaults(Collections.emptyMap());
+	 * </pre>
+	 *
+	 * Or, the following would supply a custom converter for the subject, leaving the other defaults
+	 * in place:
+	 *
+	 * <pre>
+	 * 	MappedJwtClaimsSetConverter.withDefaults(
+	 * 		Collections.singletonMap(JwtClaimNames.SUB, new UserDetailsServiceJwtSubjectConverter()));
+	 * </pre>
+	 *
+	 * To completely replace the underlying {@link Map} of converters, {@see MappedJwtClaimSetConverter(Map)}.
+	 *
+	 * @param attributeConverters
+	 * @return An instance of {@link MappedJwtClaimSetConverter} that contains the converters provided,
+	 *   plus any defaults that were not overridden.
+	 */
+	public static MappedJwtClaimSetConverter withDefaults
+			(Map<String, Converter<Object, ?>> attributeConverters) {
+		Assert.notNull(attributeConverters, "attributeConverters cannot be null");
+
+		Map<String, Converter<Object, ?>> attributeNameToConverter = new HashMap<>();
+		attributeNameToConverter.put(JwtClaimNames.AUD, AUDIENCE_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.EXP, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.IAT, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.ISS, ISSUER_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.JTI, STRING_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.NBF, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.SUB, STRING_CONVERTER);
+		attributeNameToConverter.putAll(attributeConverters);
+
+		return new MappedJwtClaimSetConverter(attributeNameToConverter);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Map<String, Object> convert(Map<String, Object> claims) {
+		Assert.notNull(claims, "claims cannot be null");
+
+		Map<String, Object> mappedClaims = new HashMap<>(claims);
+
+		for (Map.Entry<String, Converter<Object, ?>> entry : this.attributeConverters.entrySet()) {
+			Converter<Object, ?> converter = this.attributeConverters.get(entry.getKey());
```

Comment:
entry.getKey() is used in multiple places and is not very readable. Please assign to something like attributeName